### DISCUSSION TITLE
zh/Q-and-A/answer/question2

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#915] [Announcements]
# zh/Q-and-A/answer/question2

为什么按步骤操作了之后，运行的时候总是会出现“生成已完成，但出现错误”？ 问题描述： 我遇到了用vscode打出helloworld调试运行出现错误，显示“生成已完成，但出现错误”的问题百度之后发现看不懂也不知道怎么解决，要如何处理？

https://cygnomatic-controller.github.io/zh/Q-and-A/answer/question2.html

**Comment by GzqHerry** [comment]
你是否有成功编译过？如果没有，请详细按照vscode安装文档配置

**Reply by GzqHerry** [reply]
运行的时候点击的是右上角的小三角形，小三角形可能会有一个小甲虫。点击之后vscode正中间上方的窗口弹下一个下拉栏，上面让你选择gcc，点击确定。这个时候才是正常的

**Comment by WangJie-Mant** [comment]
cl.exe是专有于微软的编译器，而g++是开源的。虽然不知道这是否是造成上述“开发人员”问题的原因，但可以试着将编译器调成g++:
在vsc的文件项目里有个.vscode的文件夹，里面应该有至少两个配置文件：launch.json和tasks.json。可以试着编辑配置文件（担心的话就拷贝一份）将编译器调成g++:
1、按下 Ctrl + Shift + D 打开调试面板，然后点击“创建一个 launch.json 文件”。
选择 C++ (GDB/LLDB)。
以下是launch.json的相关配置。
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug",
            "type": "cppdbg",
            "request": "launch",
            "program": "${fileDirname}/${fileBasenameNoExtension}.exe",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "setupCommands": [
                {
                    "text": "-enable-pretty-printing",
                    "description": "enable pretty printing",
                    "ignoreFailures": true
                }
            ]
        }
    ]
}
2、打开你的工作区或项目文件夹。
按下 Ctrl + Shift + P，输入并选择 “Tasks: Configure Default Build Task”。
如果没有现成的 tasks.json，选择创建一个新的。
以下是tasks.json的内容。
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "build",
            "type": "shell",
            "command": "g++",
            "args": [
                "-g",                // 生成调试信息
                "${file}",          // 当前打开的文件
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}.exe" // 输出文件路径
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "problemMatcher": ["$gcc"]
        }
    ]
}
3、记得将你的minggw文件夹的bin文件夹路径放在系统环境变量里面。
4、有个推荐的插件叫code runner，可以一试。可以不用右上角的运行，直接ctrl+shift+b进行conpile+build，用ctrl+alt+n来运行。
5、请一定将你原先的配置文件备份。每个人配置的过程不同，很难一招吃遍天下鲜~

**Reply by 477996850** [reply]
1.有可能代码中存在语法错误或者逻辑错误，可以检查代码并修复错误
2.生成器配置错误或者不完整，需要检查一下生成器配置并且确保他们完整且正确
3生成器与其他插件或者扩展程序不兼容，需禁用某个插件或者运行程序并重新运行生成器

**Reply by 477996850** [reply]
不好意思回复到错误的地方了

**Comment by 477996850** [comment]
1.有可能代码中存在语法错误或者逻辑错误，可以检查代码并修复错误
2.生成器配置错误或者不完整，需要检查一下生成器配置并且确保他们完整且正确
3生成器与其他插件或者扩展程序不兼容，需禁用某个插件或者运行程序并重新运行生成器
